Plugins/ExaTrkX/CMakeLists.txt

```diff
@@ -6,9 +6,11 @@ if(ACTS_EXATRKX_ENABLE_CUDA)
 endif()
 
 if(ACTS_EXATRKX_ENABLE_ONNX)
-    target_sources(
+    target_sources(ActsPluginExaTrkX PRIVATE src/OnnxEdgeClassifier.cpp)
+    target_link_libraries(ActsPluginExaTrkX PRIVATE onnxruntime::onnxruntime)
+    target_compile_definitions(
         ActsPluginExaTrkX
-        PRIVATE src/OnnxEdgeClassifier.cpp src/OnnxMetricLearning.cpp
+        PUBLIC ACTS_EXATRKX_ONNX_BACKEND
```

Comment:
This was already present before, but let me ask anyway: I'm not a big fan of definitions like this, especially `PUBLIC` ones. Is there a way that we can avoid this?

Reply:
If we decide we should restructure this, I would prefer doing it in a follow-up PR

Reply:
ok